gdal_subsetter/transform.py

```diff
@@ -170,7 +148,55 @@ def process_item(self, item, source):
         finally:
             rmtree(output_dir)
 
-    def process_geotiff(self, source, basename, input_filename, output_dir, layernames):
+    def stage_and_get_output_stac(self, transformed_filename: str,
```

Comment:
I was thinking about this method some more - I could definitely be convinced to split it into two pieces: one that just stages the output and one that creates the STAC record. That might be a little cleaner.

Reply:
I think we can only if we also pop the call to `HarmonyAdapter.reformat` inside the method, too, because we need to call `HarmonyAdapter.get_bbox_lonlat` on the file prior to re-formating. Totally doable, but I did wonder if that was a little outside of the implied scope of the new method.

Reply:
why do you need to call it before reformat? Oh, because the transformed filename is an param. Ok. nevermind I guess.

Reply:
Yeah - part of the problem is that `HarmonyAdapter.get_bbox_lonlat` doesn't work on all file formats. So if the final output format is NetCDF-4, then `HarmonyAdapter.get_bbox_lonlat` would fail (I think).